src/Common/src/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -60,37 +59,25 @@ private static Marshaller[] InitializeMarshallers(PInvokeMethodData pInvokeMetho
 
         private MethodIL EmitIL()
         {
-            // We have 4 code streams:
-            // - _marshallingCodeStream is used to convert each argument into a native type and 
-            // store that into the local
-            // - callsiteSetupCodeStream is used to used to load each previously generated local
-            // and call the actual target native method.
-            // - _returnValueMarshallingCodeStream is used to convert the native return value 
-            // to managed one.
-            // - _unmarshallingCodestream is used to propagate [out] native arguments values to 
-            // managed ones.
-
-            ILEmitter emitter = new ILEmitter();
-            ILCodeStream fnptrLoadStream = emitter.NewCodeStream();
-            ILCodeStream marshallingCodeStream = emitter.NewCodeStream();
-            ILCodeStream callsiteSetupCodeStream = emitter.NewCodeStream();
-            ILCodeStream returnValueMarshallingCodeStream = emitter.NewCodeStream();
-            ILCodeStream unmarshallingCodestream = emitter.NewCodeStream();
+            PInvokeILCodeStreams pInvokeILCodeStreams = new PInvokeILCodeStreams();
+            ILEmitter emitter = pInvokeILCodeStreams.Emitter;
+            ILCodeStream fnptrLoadStream = pInvokeILCodeStreams.FunctionPointerLoadStream;
+            ILCodeStream callsiteSetupCodeStream = pInvokeILCodeStreams.CallsiteSetupCodeStream;
+            ILCodeStream unmarshallingCodestream = pInvokeILCodeStreams.UnmarshallingCodestream;
 
             // Marshal the arguments
             for (int i = 0; i < _marshallers.Length; i++)
             {
-                Marshaller marshaller = _marshallers[i];
-                marshaller.EmitMarshallingIL(emitter, marshallingCodeStream, callsiteSetupCodeStream, unmarshallingCodestream, returnValueMarshallingCodeStream);
+                _marshallers[i].EmitMarshallingIL(pInvokeILCodeStreams);
             }
 
             // make the call
-            TypeDesc nativeReturnType = _marshallers[0].NativeType;
+            TypeDesc nativeReturnType = _marshallers[0].UnsafeType;
```

Comment:
Agreed. This was done to have parity with .NET Native. I will revert back to NativeType/ManagedType.